learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -128,7 +139,11 @@ async def social_handler(request):
             "We only handle Google logins. Non-google Provider"
         )
 
-    user = await _google(request)
+    user = await _handle_google_authorization(request)
```

Comment:
This function ought to handle a single option of signing in for a given API.

When we create the endpoints for Oauth, we ought to initialized them separately.

For example in Routes, where we initialize the google oauth route, we should use
```python
    # Copied from learning_observer/routes.py:239
    if 'google_oauth' in settings.settings['auth']:
        debug_log("Running with Google authentication")
        app.add_routes([
            aiohttp.web.get(
                '/auth/login/{provider:google}',
                # the below call to `social_handler` ought to pass a parameter "google"
                handler=learning_observer.auth.social_handler),
        ])
```

For the canvas oauth routes, we should add a similar route but with
`handler=learning_observer.auth.social_handler('canvas')` or similar.

The passed method should call the appropriate authorization function instead of always calling Google and sometimes calling Canvas/other LMSs